src/finch/interface/lazy.py

```diff
@@ -327,7 +337,173 @@ def prod(arr: LazyTensor, dims) -> LazyTensor:
     .. versionchanged:: 2023.12
        Required the function to return a floating-point array having the same data type as the input array when provided a floating-point array.
     """
-    return reduce(operator.mul, arr, dims, arr.fill_value)
+    return reduce(operator.mul, arr, axis=axis, keepdims=keepdims, init=arr.fill_value)
 
 def multiply(x1: LazyTensor, x2: LazyTensor) -> LazyTensor:
-    return elementwise(operator.mul, x1, x2)
\ No newline at end of file
+    return elementwise(operator.mul, x1, x2)
+
+def sum(
+        arr: LazyTensor,
+        /,
+        *,
+        axis: int | tuple[int, ...] | None = None,
+        dtype = None,
+        keepdims: bool = False
+    ) -> LazyTensor:
+    """
+    Calculates the sum of input array ``arr`` elements.
+    """
+    return reduce(operator.add, arr, axis=axis, keepdims=keepdims, init=arr.fill_value)
```

Comment:
You should add `init=None` in the function signature, and pass the value of init to `reduce`, which will query the `initial_value` function in `algebra.py` to figure out what the initial value should be. Do this for all the reduction functions. You'll need to add initial value properties for most of these.